File: cryze_android_app/app/src/main/java/com/tencentcs/iotvideo/messagemgr/IMessageMgr.java
package com.tencentcs.iotvideo.messagemgr;

import com.tencentcs.iotvideo.utils.rxjava.IResultListener;
import java.util.concurrent.ConcurrentHashMap;
/* loaded from: classes2.dex */
public interface IMessageMgr {
    void addAppLinkListener(IAppLinkListener iAppLinkListener);

    void addEventListener(IEventListener iEventListener);

    void addModelListener(IModelListener iModelListener);

    void notifyDevModelToServer(String str, String str2);

    void operateProUser(String str, String str2, String str3, String str4, int i10, IResultListener<ModelMessage> iResultListener);

    void operateProUser(String str, String str2, String str3, String str4, IResultListener<ModelMessage> iResultListener);

    void readProperty(String str, String str2, int i10, IResultListener<ModelMessage> iResultListener);

    void readProperty(String str, String str2, IResultListener<ModelMessage> iResultListener);

    void removeAppLinkListener(IAppLinkListener iAppLinkListener);

    void removeAppLinkListeners();

    void removeEventListener(IEventListener iEventListener);

    void removeEventListeners();

    void removeModelListener(IModelListener iModelListener);

    void removeModelListeners();

    void sendDataToDevice(String str, byte b10, byte[] bArr, boolean z10, int i10, IResultListener<DataMessage> iResultListener);

    void sendDataToDevice(String str, byte[] bArr, boolean z10, int i10, IResultListener<DataMessage> iResultListener);

    void sendDataToDevice(String str, byte[] bArr, boolean z10, int i10, boolean z11, IResultListener<DataMessage> iResultListener);

    void sendDataToDeviceWithResponse(String str, byte b10, byte[] bArr, IResultListener<DataMessage> iResultListener);

    void sendDataToDeviceWithResponse(String str, byte[] bArr, IResultListener<DataMessage> iResultListener);

    void sendDataToDeviceWithoutResponse(String str, byte b10, byte[] bArr, IResultListener<DataMessage> iResultListener);

    void sendDataToDeviceWithoutResponse(String str, byte[] bArr, IResultListener<DataMessage> iResultListener);

    void sendDataToServer(String str, byte[] bArr, int i10, IResultListener<DataMessage> iResultListener);

    void sendDataToServer(String str, byte[] bArr, IResultListener<DataMessage> iResultListener);

    void setPassThroughListener(IPassThroughListener iPassThroughListener);

    void subscribeDevice(ConcurrentHashMap<Integer, IResultListener<Boolean>> concurrentHashMap);

    void writeProperty(String str, String str2, String str3, int i10, IResultListener<ModelMessage> iResultListener);

    void writeProperty(String str, String str2, String str3, IResultListener<ModelMessage> iResultListener);
}

File: cryze_android_app/android/app/src/main/java/com/tencentcs/iotvideo/messagemgr/IMessageMgr.kt
package com.tencentcs.iotvideo.messagemgr

import com.tencentcs.iotvideo.utils.rxjava.IResultListener
import java.util.concurrent.ConcurrentHashMap

//TODO: pick through the JNI libraries and whittle this down to what's needed.
interface IMessageMgr {
    fun addAppLinkListener(iAppLinkListener: IAppLinkListener?)

    fun addEventListener(iEventListener: IEventListener?)

    fun addModelListener(iModelListener: IModelListener?)

    fun notifyDevModelToServer(str: String?, str2: String?)

    fun operateProUser(
        str: String?,
        str2: String?,
        str3: String?,
        str4: String?,
        i10: Int,
        iResultListener: IResultListener<ModelMessage?>?
    )

    fun operateProUser(
        str: String?,
        str2: String?,
        str3: String?,
        str4: String?,
        iResultListener: IResultListener<ModelMessage?>?
    )

    fun readProperty(
        str: String?,
        str2: String?,
        i10: Int,
        iResultListener: IResultListener<ModelMessage?>?
    )

    fun readProperty(str: String?, str2: String?, iResultListener: IResultListener<ModelMessage?>?)

    fun removeAppLinkListener(iAppLinkListener: IAppLinkListener?)

    fun removeAppLinkListeners()

    fun removeEventListener(iEventListener: IEventListener?)

    fun removeEventListeners()

    fun removeModelListener(iModelListener: IModelListener?)

    fun removeModelListeners()

    fun sendDataToDevice(
        str: String?,
        b10: Byte,
        bArr: ByteArray?,
        z10: Boolean,
        i10: Int,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun sendDataToDevice(
        str: String?,
        bArr: ByteArray?,
        z10: Boolean,
        i10: Int,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun sendDataToDevice(
        str: String?,
        bArr: ByteArray?,
        z10: Boolean,
        i10: Int,
        z11: Boolean,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun sendDataToDeviceWithResponse(
        str: String?,
        b10: Byte,
        bArr: ByteArray?,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun sendDataToDeviceWithResponse(
        str: String?,
        bArr: ByteArray?,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun sendDataToDeviceWithoutResponse(
        str: String?,
        b10: Byte,
        bArr: ByteArray?,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun sendDataToDeviceWithoutResponse(
        str: String?,
        bArr: ByteArray?,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun sendDataToServer(
        str: String?,
        bArr: ByteArray?,
        i10: Int,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun sendDataToServer(
        str: String?,
        bArr: ByteArray?,
        iResultListener: IResultListener<DataMessage?>?
    )

    fun subscribeDevice(concurrentHashMap: ConcurrentHashMap<Int, IResultListener<Boolean>>?)

    fun writeProperty(
        str: String?,
        str2: String?,
        str3: String?,
        i10: Int,
        iResultListener: IResultListener<ModelMessage?>?
    )

    fun writeProperty(
        str: String?,
        str2: String?,
        str3: String?,
        iResultListener: IResultListener<ModelMessage?>?
    )
}
